src/dbnode/encoding/proto/custom_unmarshal.go

```diff
@@ -68,18 +68,15 @@ func (u *customUnmarshaler) numNonCustomValues() int {
 	return u.numNonCustom
 }
 
-func (u *customUnmarshaler) nonCustomFieldValues() *dynamic.Message {
-	if u.nonCustomValues == nil {
-		u.nonCustomValues = dynamic.NewMessage(u.schema)
-	}
+func (u *customUnmarshaler) nonCustomFieldValues() []marshaledField {
 	return u.nonCustomValues
 }
 
 func (u *customUnmarshaler) unmarshal() error {
 	u.customValues = u.customValues[:0]
 
 	if u.nonCustomValues != nil {
-		u.nonCustomValues.Reset()
+		u.nonCustomValues = u.nonCustomValues[:0]
```

Comment:
Do you need to zero out each entry to avoid holding onto refs or does `marshalledField` here not contain references to longly lived pointer types?

Reply:
Hm I see a `[]byte` in that struct that could be longly held onto.